lib/ansible/plugins/lookup/password.py

```diff
@@ -73,91 +77,149 @@ def _parse_parameters(term):
     params['chars'] = params.get('chars', None)
     if params['chars']:
         tmp_chars = []
-        if ',,' in params['chars']:
+        if u',,' in params['chars']:
             tmp_chars.append(u',')
-        tmp_chars.extend(c for c in params['chars'].replace(',,', ',').split(',') if c)
+        tmp_chars.extend(c for c in params['chars'].replace(u',,', u',').split(u',') if c)
         params['chars'] = tmp_chars
     else:
         # Default chars for password
-        params['chars'] = ['ascii_letters', 'digits', ".,:-_"]
+        params['chars'] = [u'ascii_letters', u'digits', u".,:-_"]
 
     return relpath, params
 
+def _random_password(length=DEFAULT_LENGTH, chars=C.DEFAULT_PASSWORD_CHARS):
+    '''
+    Return a random password string of length containing only chars.
+    NOTE: this was moved from the old ansible utils code, as nothing
+          else appeared to use it.
+    '''
+
+    # hate how python implements asserts. Our text strategy says that we
+    # should never be passing byte strings around unless the function
+    # explicitly takes a byte string as its argument.  Therefore if we get
+    # a byte string here it means that the problem is in the calling code.
+    # Asserts would be perfect for this if they were disabled by default.
+    # Alas, in python, they're active by default; only turned off if
+    # python is run with optimization.
+    # assert isinstance(chars, text_type)
+    random_generator = random.SystemRandom()
+
+    password = []
+    while len(password) < length:
+        new_char = random_generator.choice(chars)
+        if new_char in chars:
+            password.append(new_char)
+
+    return u''.join(password)
+
+def _random_salt():
+    salt_chars = to_text(ascii_letters + digits + './', errors='surrogate_or_strict')
+    return _random_password(length=8, chars=salt_chars)
+
+def _gen_candidate_chars(characters):
+    chars = []
+    for chars_spec in characters:
+        chars.append(to_text(getattr(string, to_native(chars_spec), chars_spec),
+                            errors='surrogate_or_strict'))
+    chars = u''.join(to_text(c) for c in chars).replace(u'"', u'').replace(u"'", u'')
+    return chars
+
+
+def _gen_password(length, chars):
+    chars = _gen_candidate_chars(chars)
+    password = u''.join(random.choice(chars) for _ in range(length))
+    return password
+
+def _create_password_file(b_path):
```

Comment:
I haven't figured out a good way to split _read_or_create_file better... The logic of the module is a little odd in that regards.

read_or_create_password_file() is sort of get_to_the_state_where_we_are_ready_to_write_the_password_file()

It's crammed into one method so that later code can avoid having to deal with the two potential scenarios. And, or, avoid  abstractions like

``` python

pw_store =  load_password_store()
pw_data = pw_store.read()
pw_data = update_pw_data(pw_data)
pw_store.update(pw_data)
pw_store.save()
```